### ISSUE TITLE
Examples: amp-base-carousel: Redundant/unnecessary keyboard focus stops

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31165]
## WCAG Level
Level A

## Priority
Medium

## Type
Navigation

## Pages/screens/Components affected
* [amp-base-carousel](https://amp.dev/documentation/components/amp-base-carousel/)

## Description
For people who navigate using a keyboard or keyboard-like device (such as a switch control or a joystick) the focus order of interactive controls in a document should be logical and intuitive.

In the [basic carousel with looping Glitch example](https://nutritious-tide-hornet.glitch.me/amp-base-carousel/captions.html), the container around the individual slides (in the shadow root) is currently focusable, despite being non-interactive.

## User impact
Duplicate/unnecessary tab stops are likely to cause irritation for keyboard-only users when they are navigating through the page because it means they have to use twice as many key presses to get where they want to go. For people using an alternative input device like a head-wand (that mimics keyboard interaction) the duplicated effort can even cause fatigue. Further, if elements that are not interactive receive focus, this may lead users to think that the elements are actionable - when attempting to activate them has no effect, this may lead to further confusion.

Currently, the demo (and `amp-base-carousel` in general) has no visible focus indication (noted in a separate issue). For a sighted keyboard user, it's unexpected that the slide container itself would receive focus - they will likely just expect the previous/next buttons to be focusable. This currently leads to an unnecessarily confusing experience.

For an assistive technology user, the experience is more disruptive - once the container receives focus, its contents are announced/read out - meaning that users will hear all slides, announced as one single uninterrupted stream.

## Required solution
Make sure that only actionable controls fall into the regular keyboard focus order.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance
If arbitrary elements (such as `<div>` containers or similar) have been made keyboard focusable through the use of `tabindex`, remove the attribute.

Currently, the shadow root for the carousel in the demo (but, strangely, not in the other demos/examples) has the following structure:

```
<div class="scroll-container-34d526e ..." tabindex="0" style="scroll-behavior: smooth;">
  <div data-slide="1" ...> ... </div>
  <div data-slide="2" ...> ... </div>
  <div data-slide="0" ...> ... </div>
</div>
```

Remove the `tabindex="0"` attribute from this container.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Use `Tab`/`Shift`+`Tab` to navigate through the page.
2. Check that non-interactive/non-functional controls don't fall into the focus order.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG21/#focus-order)

## Test data
Test date: November 2020
